commons-geometry-spherical/src/main/java/org/apache/commons/geometry/spherical/twod/RegionBSPTree2S.java

```diff
@@ -171,16 +171,32 @@ public RegionBSPTree2S toTree() {
         final DoublePrecisionContext precision = ((GreatArc) getRoot().getCut()).getPrecision();
 
         double sizeSum = 0;
-        Vector3D centroidVector = Vector3D.ZERO;
+        Vector3D centroidVectorSum = Vector3D.ZERO;
+
+        Vector3D centroidVector;
```

Comment:
Any reason the declaration is not inside the loop? It doesn't look to be used elsewhere.